com.unity.perception/Runtime/GroundTruth/Labelers/SemanticSegmentationLabeler.cs

```diff
@@ -192,6 +196,37 @@ protected override void Setup()
             visualizationEnabled = supportsVisualization;
         }
 
+        private void SetupVisualizationElements()
+        {
+            segmentTransparency = 0.8f;
+            backgroundTransparency = 0.0f;
+
+            segVisual = GameObject.Instantiate(Resources.Load<GameObject>("SegmentTexture"));
+
+            segImage = segVisual.GetComponent<RawImage>();
+            segImage.material.SetFloat("_SegmentTransparency", segmentTransparency);
+            segImage.material.SetFloat("_BackTransparency", backgroundTransparency);
+            segImage.texture = targetTexture;
+
+            var rt = segVisual.transform as RectTransform;
+            rt.SetSizeWithCurrentAnchors(RectTransform.Axis.Horizontal, camWidth);
+            rt.SetSizeWithCurrentAnchors(RectTransform.Axis.Vertical, camHeight);
+
+            if (segCanvas == null)
```

Comment:
OK. I'm curious, did you try https://docs.unity3d.com/ScriptReference/GUI.DrawTexture.html ?

Reply:
Oh yeah, spent a whole day fighting with that and with Graphics.DrawTexture. So the problem with GUI.DrawTexture is that you can not pass in a material with it, so there was no way to associate our shader to the texture. This brought back the problem with the whole screen being black again.

So, I also went down the road with Graphics.DrawTexture, which is similar to GUI.DrawTexture, but allows you to pass in a material. Unfortunately, when I used this call I didn't have control on where it was drawn to the screen, so it could be drawn on top of all of the UI components. So, if you were to use the slider and set the background to completely opaque, it could (and consistently in my tests) cover over the UI and bounding boxes. Added to the top of that, it also felt pretty hacky.

So, I begrudgingly, kept our old technique for this special case.

Reply:
Got it. So for future thinking, what you could probably do is create a second RenderTexture and use Graphics.Blit to render the segmentation into that texture using your material and then call GUI.DrawTexture(). But this is fine I think - shouldn't harm anything.

Reply:
Ok, I'll try this in the future